### PR TITLE
health-check and storage-metrics support for fluent-bit

### LINKED ISSUE #1286: Add support to enable health-check and storage-metrics for fluent-bit in loggings-file
Hi their, so I tell you what's about . . . 

**Is your feature request related to a problem? Please describe.**
We're using logging-operator for quiet sometime and lately sometimes we notice fluentd is not receiving any logs from fluent-bit pods without a noticable reason. Redeploying is fixing it, but liveness- and readiness probes could handle this without any user interaction. 

Fluent-bit can provide health information and additional storage metrics to monitor. Both can be enabled in fluent-bit.conf, but logging-operator reverts changes in the corresponding secret as we save it. I've done some test with logging-operator scaled down.

**Describe the solution you'd like**
I'd like to enable health-check and storage-metrics for fluent-bit via logging-operators logging-file. And then logging-operator sets "Health_Check" and "storage.metrics" in the secret that's mounted as fluent-bit.conf.

**Describe alternatives you've considered**
There seem's to be not really an alternative to get similiar metrics from the fluent-bits managed by logging-operator I can think of.

**Additional context**
- The health-check in fluent-bit I've mentioned is described here: https://docs.fluentbit.io/manual/administration/monitoring#health-check-for-fluent-bit
- The storage-metrics option is described here at the end of the table: https://docs.fluentbit.io/manual/administration/buffering-and-storage#service-section-configuration

### PR DESCRIPTION
**Pull request by nak0f**
PR for adding support to enable health-check and storage-metrics for fluent-bit. (fixes #1286)

### CODE REVIEW

**Comment by OverOrion** [comment]
Thank you for your contribution! The CI fails because there are generated files which had not been pushed. Could you please add them as well? `make check-diff` should tell you what the difference is, currently it seems that `docs/configuration/crds/v1beta1/fluentbit_types.md` is the culprit.

**Comment by nak0f** [comment]
> Thank you for your contribution! The CI fails because there are generated files which had not been pushed. Could you please add them as well? `make check-diff` should tell you what the difference is, currently it seems that `docs/configuration/crds/v1beta1/fluentbit_types.md` is the culprit.

Thanks @OverOrion :) That should be sorted out now.

File: pkg/resources/fluentbit/config.go [hunk 1]
```diff
@@ -42,6 +42,22 @@ var fluentBitConfigTemplate = `
     {{ $key }}  {{$value}}
     {{- end }}
     {{- end }}
+    {{- if .HealthCheck }}
+    {{- if .HealthCheck.Enabled }}
+    Health_Check On
+    {{- else }}
+    Health_Check Off
+    {{- end }}
```
Comment (pepov) [inline pkg/resources/fluentbit/config.go:50]: does it make sense to have a specific flag for this? wouldn't we want to enable healthcheck always if the healthcheck struct is defined?

Comment (nak0f) [reply pkg/resources/fluentbit/config.go]: I just replicated the same config structure from fluent-bit but fair point. From a logging-operator user perspective this flag is not really needed.

**Review by pepov — APPROVED** [review]

**Review by OverOrion — APPROVED** [review]
LGTM, thank you once again for your contribution!

**Pull request merged**